File: Data Structures & Algorithms/products-of-array-discluding-self/submission-7.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int prefix = 1;
        int posfix = 1;
        int[] res = new int[nums.length];

        for (int i = 0; i < nums.length; i++) {
            res[i] = prefix;
            prefix *= nums[i];
        }

        for (int j = nums.length - 1; j >= 0; j--) {
            res[j] *= posfix;
            posfix *= nums[j];
        }
        return res;
    }
}  

File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        freqs = [[] for i in range(len(nums) + 1)]
        res = []

        for n in nums:
            count[n] = 1 + count.get(n, 0)
        for i, n in count.items():
            freqs[n].append(i)
        
        for i in range(len(freqs) - 1, -1, -1):
            for freq in freqs[i]:
                res.append(freq)
                if len(res) == k:
                    return res
